### DISCUSSION TITLE
Hangs forever when trying to add the integration

### DISCUSSION THREAD

**Discussion by llucax** [Tasshack/dreame-vacuum#353] [Q&A] [answered]
Hi, I just followed the manual install steps and the install was successful. I restarted HA and when I click on add integration, it hangs forever with the message "Please wait, starting configuration wizard for dreame_vacuum".

![image](https://github.com/Tasshack/dreame-vacuum/assets/1031485/fe61165c-81ef-440b-994e-abc83c018e30)

Any ideas of what could be going on and how to fix it?

**Comment by llucax** [comment]
In case it is relevant, I'm using the HA container install.

**Reply by Tasshack** [reply]
Probably HA cannot install some dependency and you should check the logs to see that. Also please make sure you have reserved more than 2GB memory for the container.

**Reply by llucax** [reply]
Hi, I don't see anything in the logs, just this:

```
Dec 31 01:14:30 rock64 sh[500686]: services-up: info: copying legacy longrun home-assistant (no readiness notification)
Dec 31 01:14:30 rock64 sh[500686]: s6-rc: info: service legacy-services successfully started
Dec 31 01:14:35 rock64 sh[500686]: 2023-12-31 01:14:35.922 WARNING (SyncWorker_4) [homeassistant.loader] We found a custom integration dreame_vacuum which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
Dec 31 01:14:42 rock64 sh[500686]: 2023-12-31 01:14:42.563 WARNING (Recorder) [homeassistant.components.recorder.util] The system could not validate that the sqlite3 database at //config/home-assistant_v2.db was shutdown cleanly
Dec 31 01:14:42 rock64 sh[500686]: 2023-12-31 01:14:42.792 WARNING (Recorder) [homeassistant.components.recorder.util] Ended unfinished session (id=11 from 2023-12-30 23:32:44.944872)
```

The DB error is from before I installed this integration, and I looked for info and didn't find much useful info of why it could happen, but so far I didn't experience anything wrong.

About memory, the container is not limited at all AFAIK, I'm just using `docker run` without passing any limits, the box is a rock64 with 4GB and this is the memory status:

```console
rock64:home-assistant:% free -m                         
               total        used        free      shared  buff/cache   available
Mem:            3974         631         424           5        3008        3342
Swap:           1987           1        1985
```

If you have any other suggestions on how to debug this it would be much appreciated. Thanks!

**Reply by llucax** [reply, marked as answer]
In case anybody is reading, after many many failed attempts I decide to try to debug this. I enabled debug logging **globally* (which produces a **lot** of logs) and after that the wizard just worked. Not sure if it was a coincidence or if there is some sort of race and enabling logging makes everything slower and makes the race go away.

After I completed the setup, I disabled debug logging and everything seems to be working fine (although I didn't spend time yet configuring the cards and actually using the robot).

**Reply by Ives25** [reply]
Hi,
I installed the latest beta version for my L20 Ultra (HA on VM Proxmox). The installation goes normally but then my Home Assistant disconnects. It reconnects for a short while then disconnects again, etc.
I had to restore a backup of the VM to have HA again.

**Reply by Tasshack** [reply]
@Ives25 this may be a on Proxmox hardware. https://github.com/Tasshack/dreame-vacuum/issues/401

**Reply by Ives25** [reply]
Thank you for your reply.
The difficulty is that after installing this integration, I lose control of HA and I am forced to restore a Proxmox VM with the consequence of losing data for consumption histories.
Can we expect a fix in a next stable version?

**Reply by llucax** [reply]
@Ives25 this looks like a completely different issue, it might be better to open a separate discussion for it, otherwise you are notifying people with stuff unrelated to this discussion.

**Reply by Ives25** [reply]
@llucax 
Before the last HA update, I couldn't finish installing the integration with HA blocking, which is the subject of this discussion. Since the last HA update, loss of HA control happens after installation.
